### ISSUE TITLE
wp_actionscheduler_actions and wp_actionscheduler_logs db tables are over 55GB in size

### ISSUE THREAD

**Issue by lukecav** [woocommerce/woocommerce#26818]
Site is running the current version of WooCommerce 4.2.0 but both of the Action Scheduler db tables for actions and logs have bloated in size.

`wp_actionscheduler_actions` -- **58.3 GB** rows 137,226,929
`wp_actionscheduler_logs` -- **55.7 GB** rows 412,139,259

**Comment by lukecav** [comment]
```
affiliate-wp    active  none    2.5.4
akismet active  none    4.1.6
astra-addon active  available   2.2.5
astra-widgets   active  available   1.2.3
cartflows   active  none    1.5.11
cartflows-pro   active  available   1.5.2
elementor   active  available   2.9.8
elementor-pro   active  available   2.9.4
webtoffee-gdpr-cookie-consent   active  version higher than expected    2.2.9
ithemes-security-pro    active  none    6.6.0
kadence-woocommerce-email-designer  active  none    1.4.1
postmark-approved-wordpress-plugin  active  none    1.12.4
shortpixel-image-optimiser  active  none    4.19.2
swift-performance   active  none    2.1.8
woocommerce active  none    4.2.0
woo-cart-abandonment-recovery   active  none    1.2.7
wc-dynamic-pricing-and-discounts    active  none    2.3.8
force-default-variant-for-woocommerce   active  none    1.5.2
woocommerce-gateway-paypal-express-checkout active  none    2.0.2
woocommerce-photo-reviews   active  available   1.1.4.2
woocommerce-product-price-based-on-countries    active  none    2.0.8
woocommerce-price-based-country-pro-addon   active  none    2.8.14
twist-cc    active  none    2.1.0.1
woocommerce-sequential-order-numbers-pro    active  none    1.15.3
woocommerce-gateway-stripe  active  none    4.4.0
wp-crontrol active  none    1.8.5
wp-fusion   active  available   3.33.10
wp-fusion-media-tools   active  none    1.1
WPShapere   active  none    6.1.2
wp-user-avatar  active  none    2.2.7
wpseo-woocommerce   active  none    13.2
```

**Comment by lukecav** [comment]
Currently, do AS logs purge after a set number of days automatically?

**Comment by lukecav** [comment]
wp_actionscheduler_actions |   | Data: 36413.34MB + Index: 23264.52MB + Engine InnoDB

**Comment by lukecav** [comment]
![AS Complete actions](https://user-images.githubusercontent.com/5797332/85062623-095b9980-b166-11ea-9d95-39229061fe90.png)

**Comment by lukecav** [comment]
https://wordpress.org/support/topic/removing-wc-admin_import_orders-wc-admin_import_customers/

**Comment by lukecav** [comment]
A number of the flood of actions created related to `wc-admin_import_orders`

**Comment by lkraav** [comment]
Probably

https://github.com/woocommerce/action-scheduler/issues/517
https://github.com/woocommerce/action-scheduler/issues/541
https://github.com/woocommerce/woocommerce-admin/issues/4255

**Comment by lukecav** [comment]
I was able to empty out both of the Action Scheduler db tables for actions and logs and then set the retention for AS to be one week. The site database is back down to 567.68MB it was at 117GB in size before the purge.

**Comment by lukecav** [comment]
Current db table sizes.
wp_actionscheduler_actions |   | Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_logs |   | Data: 0.02MB + Index: 0.03MB + Engine InnoDB

**Comment by rrennick** [comment]
@lukecav I'll close this as https://github.com/woocommerce/woocommerce-admin/issues/4255 created the flood of import orders when there was an order that caused an error on import. That issue was fixed in WC 4.2.
